expr/functions/integralByInterval/function.go

```diff
@@ -0,0 +1,108 @@
+package integralByInterval
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/bookingcom/carbonapi/expr/helper"
+	"github.com/bookingcom/carbonapi/expr/interfaces"
+	"github.com/bookingcom/carbonapi/expr/types"
+	"github.com/bookingcom/carbonapi/pkg/parser"
+	dataTypes "github.com/bookingcom/carbonapi/pkg/types"
+)
+
+type integralByInterval struct {
+	interfaces.FunctionBase
+}
+
+func GetOrder() interfaces.Order {
+	return interfaces.Any
+}
+
+func New(configFile string) []interfaces.FunctionMetadata {
+	res := make([]interfaces.FunctionMetadata, 0)
+	f := &integralByInterval{}
+	functions := []string{"integralByInterval"}
+	for _, n := range functions {
+		res = append(res, interfaces.FunctionMetadata{Name: n, F: f})
+	}
+	return res
+}
+
+// integralByInterval(seriesList, intervalString)
+func (f *integralByInterval) Do(e parser.Expr, from, until int32, values map[parser.MetricRequest][]*types.MetricData) ([]*types.MetricData, error) {
```

Comment:
Please, add a comment in the appropriate form.

Reply:
Yes, I also think there should be comments here. But I didn't find examples in other functions.

Reply:
Sorry, we have missing comments in many places. Feel free to fix that)